README.md

```diff
@@ -85,45 +85,117 @@ If running on MacOS:
 Usage: databroker-perf [OPTIONS]
 
 Options:
-  -i, --iterations <ITERATIONS>  Number of iterations to run [default: 1000]
-      --api <API>                Api of databroker [default: kuksa.val.v1] [possible values: kuksa.val.v1, sdv.databroker.v1]
-      --host <HOST>              Host address of databroker [default: http://127.0.0.1]
-      --port <PORT>              Port of databroker [default: 55555]
-      --skip <ITERATIONS>        Number of iterations to run (skip) before measuring the latency [default: 10]
-      --interval <MILLISECONDS>  Minimum interval in milliseconds between iterations [default: 0]
-      --config <FILE>            Path to configuration file
-      --run-forever              Run the measurements forever (until receiving a shutdown signal)
-  -v, --verbosity <LEVEL>        Verbosity level. Can be one of ERROR, WARN, INFO, DEBUG, TRACE [default: WARN]
-  -h, --help                     Print help
-  -V, --version                  Print version
+  -r, --run-seconds <RUN_SECONDS>  Number of seconds to run [default: 8]
+      --api <API>                  Api of databroker [default: kuksa.val.v1] [possible values: kuksa.val.v1, kuksa.val.v2, sdv.databroker.v1]
+      --host <HOST>                Host address of databroker [default: http://127.0.0.1]
+      --port <PORT>                Port of databroker [default: 55555]
+      --skip-seconds <ITERATIONS>  Seconds to run (skip) before measuring the latency [default: 4]
+      --detail-output              Print more details in the summary result
+      --config <FILE>              Path to configuration file
+      --run-forever                Run the measurements forever (until receiving a shutdown signal)
+  -v, --verbosity <LEVEL>          Verbosity level. Can be one of ERROR, WARN, INFO, DEBUG, TRACE [default: WARN]
+  -h, --help                       Print help
+  -V, --version                    Print version
 ```
 
 ```
 ./target/release/databroker-perf [OPTIONS]
 ```
 
+## Default test result output
+
+By default, the results output will be shown like this:
+```
+[00:00:08] Group: Frame A | Cycle(ms): 10 | Current latency: 0.725 ms [==============================================================================================================]       8/8       seconds
```

Comment:
Splitting the output results into groups as well seems counter-intuitive, harder to understand.

Now, the user has to do the mental work to combine and aggregate all the results of the individual groups into a single result. How would he do that? Calculating the average of all the averages?

Imho, sending signals is grouped (as seen by the provider, sort of CAN-Bus point of view), but **measuring** the latency is done on the application side (subscriber, which is on VSS level) and the subscriber doesn't really know about groups of CAN signals and cycle times.

May be a future feature request to also show the "Total summary over all groups".

Reply:
Added a new global summary with calculated average of latencies.